db/seeds.rb

```diff
@@ -8,83 +8,84 @@
   status: false,
   role_id: 3)
 
-20.times do |n|
-  name = Faker::Name.name[0..10]
-  email = "user#{n+1}@gmail.com"
-  role_id = 1
-  User.create!(name: name,
-    status: false,
-    password: "password", password_confirmation: "password",
-    role_id: role_id,
-    email: email)
-end
+# 20.times do |n|
+#   name = Faker::Name.name[0..10]
+#   email = "user#{n+1}@gmail.com"
+#   role_id = 1
+#   User.create!(name: name,
+#     status: false,
+#     password: "password", password_confirmation: "password",
+#     role_id: role_id,
+#     email: email)
+# end
 
 Category.create!(name: "Học ngoại ngữ")
-Category.create!(name: "Khoa học - Kĩ thuật")
-Category.create!(name: "Tâm Lý - Kỹ Năng Sống")
-Category.create!(name: "Văn hóa - Tôn giáo")
-Category.create!(name: "Lịch sử - Chính trị")
+# Category.create!(name: "Khoa học - Kĩ thuật")
+# Category.create!(name: "Tâm Lý - Kỹ Năng Sống")
+# Category.create!(name: "Văn hóa - Tôn giáo")
+# Category.create!(name: "Lịch sử - Chính trị")
 
+Book.create!(name: "Game of thrones", category_id: 1, status: true, description:
+  "Hay, hap dan, loi cuon", user_id: 1)
+# 20.times do |n|
+#   name = Faker::Name.name[0..10]
+#   Author.create!(name: name)
+# end
 
-20.times do |n|
-  name = Faker::Name.name[0..10]
-  Author.create!(name: name)
-end
+# 100.times do |n|
+#   name = Faker::Name.name[0..10],
+#   category_id = n%Category.count+1,
+#   user_id = n%User.count+1,
+#   status = true,
+#   description = "description#{n}"
+#   Book.create!(name: name,
+#     category_id: category_id,
+#     status: status,
+#     description: description,
+#     user_id: user_id)
+# end
 
-100.times do |n|
-  name = Faker::Name.name[0..10],
-  category_id = n%Category.count+1,
-  user_id = n%User.count+1,
-  status = true,
-  description = "description#{n}"
-  Book.create!(name: name,
-    category_id: category_id,
-    status: status,
-    description: description,
-    user_id: user_id)
-end
+# 100.times do |n|
+#   category_id = n%Category.count+1
+#   Follow.create!(user_id: n%User.count+1,
+#     category_id: category_id)
+# end
 
-100.times do |n|
-  category_id = n%Category.count+1
-  Follow.create!(user_id: n%User.count+1,
-    category_id: category_id)
-end
+# 100.times do |n|
+#   book_id = n%Book.count+1
+#   Like.create!(user_id: n%User.count+1,
+#     book_id: book_id)
+# end
 
-100.times do |n|
-  book_id = n%Book.count+1
-  Like.create!(user_id: n%User.count+1,
-    book_id: book_id)
-end
+# 10.times do |n|
+#   author_id = n%Author.count+1
+#   book_id = 1
+#   Writer.create!(author_id: author_id, book_id: book_id)
+# end
 
-10.times do |n|
-  author_id = n%Author.count+1
-  book_id = 1
-  Writer.create!(author_id: author_id, book_id: book_id)
-end
+# 10.times do |n|
```

Comment:
Sao comment lại hết thế em?

Reply:
@huongnguyenmta 
Khi em chạy file seed thì nó tạo sẵn dữ liệu trong database nên lúc em viết phần test scope em lấy dữ liệu trong database ra theo scope nó lấy hết cả dữ liệu giả đã tạo nên em không so sánh được vs các dữ liệu test em tạo ạ. Thế nên em cho vào comment luôn chị! :)

Reply:
@@ em test thì dùng db test chứ nhở, env=test mà
ở đây àm môi trường development chứ liên quan gì em?

em dùng `gem "database_cleaner"`, thêm vào group test trong Gemfile nhé
TÌm hiểu luôn tác dụng và cách hoạt động gem này em nhé

Reply:
@huongnguyenmta 
- Tác dụng: làm sạch cơ sở dữ liệu khi test case. Nó làm sạch trạng thái cho testing.
- Cách hoạt động: 
+) Transaction: DB cleaner sẽ mở một giao dịch trong mỗi test case và sẽ rollback lại sau khi hoàn thành. Transaction nên được sử dụng mặc định khi nó là chiến lược nhanh nhất.
+) Deletion: Sau khi giao dịch, muốn truy cập vào dữ liệu thì sử dụng loại này mà không chậm hơn mấy so với transaction khi dữ liệu nhỏ.
+) Truncation: chiến lược chậm nhất nhưng an toàn và ổn định nhất. Nên là tốt nhất để sử dụng capybara với javascript. Và với các thiết lập lớn khi test, khi bạn muốn tạo nhiều đối tượng với các liên kết sử dụng khóa ngoài thì sẽ dùng truncation, nó sẽ lấy một khoảng thời gian cố định bất kể dữ liệu lớn bao nhiêu.